src/lib/ERC4337/accounts/Accounts.sol

```diff
@@ -0,0 +1,199 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.13;
+
+import {Mage} from "src/Mage.sol";
+import {IAccount} from "src/lib/ERC4337/interface/IAccount.sol";
+import {IEntryPoint} from "src/lib/ERC4337/interface/IEntryPoint.sol";
+import {Operations} from "src/lib/Operations.sol";
+import {IOwnable} from "src/access/ownable/interface/IOwnable.sol";
+import {OwnableInternal} from "src/access/ownable/OwnableInternal.sol";
+import {Access} from "src/access/Access.sol";
+import {SupportsInterface} from "src/lib/ERC165/SupportsInterface.sol";
+import {ECDSA} from "openzeppelin-contracts/utils/cryptography/ECDSA.sol";
+import {IERC1271} from "openzeppelin-contracts/interfaces/IERC1271.sol";
+
+// todo move into a new IAccounts interface
+error InvalidCaller(address notEntryPoint);
+
+/// @title Station Network Accounts Manager Abstract Contract
+/// @author 👦🏻👦🏻.eth
+
+/// @dev This abstract contract provides scaffolding Station's Accounts signature validation
+/// ERC1271-compliance in combination with Mage's Permissions::EXECUTE_PERMIT system
+/// provides convenient and modular private key management on an infrastructural level
+abstract contract Accounts is Mage, IAccount, IERC1271 {
+
+    /*=============
+        STORAGE
+    ==============*/
+    
+    /// @dev This chain's EntryPoint contract address
+    /// @notice Since this address is consistent across chains, `chainId` is included
+    /// in the signature digest to prevent replay attacks
+    address internal immutable _entryPoint;
+    
+    /// @dev In case of signature validation failure, return value need not include time range
+    uint256 internal constant SIG_VALIDATION_FAILED = 1;
+
+    /*=============
+        ACCOUNTS
+    ==============*/
+
+    /// @param _entryPointAddress The contract address for this chain's ERC-4337 EntryPoint contract
+    /// Official address for the most recent EntryPoint version is `0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789`
+    constructor(address _entryPointAddress) {
+        _entryPoint = _entryPointAddress;
+        
+        // permit the EntryPoint to call `execute()` on this contract via valid UserOp.signature only
+        _addPermission(Operations.EXECUTE_PERMIT, _entryPointAddress);
```

Comment:
I think it'd be wise to make this association more immutable. We can intentionally or accidentally remove permission from the entry point which isn't great...

Reply:
Good catch! Resolved in 7ddb715